.github/workflows/main.yml

```diff
@@ -14,10 +14,15 @@ on:
     branches:
       - main
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
 jobs:
   ubuntu-multiple-pythons:
     name: ${{ matrix.os }} with Python ${{ matrix.python-version }}
     runs-on: ${{ matrix.os }}
+    timeout-minutes: 60
```

Comment:
I see this was added to all the jobs, but the commit message didn't say why 😄 what was your thought here?

Reply:
Some time ago there was an instance when a test didn’t converge (before I had access to the ‘cancel job’ button), and it was necessary to wait until GH timed out. For OSX, there are only a limited number of runners, so this created a bottle neck that took literally a day to clear out. 60 minutes is imho plenty of time, as even the slowest runners finish in less than half an hour (exception is Coverage, where I put 90 minutes).

Reply:
Added this to the commit comment.